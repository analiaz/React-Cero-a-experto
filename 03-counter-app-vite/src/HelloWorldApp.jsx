/**function App(){
    return (<h1>Hola Mundo</h1>);
}

export default App;
**/

import React from 'react'

const HelloWorldApp = () => {
  return (
    <div> Hello World App </div>
  )}

export default HelloWorldApp
