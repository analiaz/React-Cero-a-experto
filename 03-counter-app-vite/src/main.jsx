import React from 'react'
import ReactDOM from 'react-dom/client'
import { CounterApp } from './CounterApp';
import { FirstApp } from './FirstApp';
import './styles.css';
CounterApp


ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
         
        {/*<CounterApp value={897} />*/}
        <FirstApp title="Hola, Soy Milk"/>
    </React.StrictMode>
)