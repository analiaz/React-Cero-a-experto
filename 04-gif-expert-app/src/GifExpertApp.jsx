import React, { useState } from 'react'
import AddCategory from './components/AddCategory';

const GifExpertApp = () => {

    const [categories, setCategories] = useState([ 'One Punch', 'Dragon ball']);

    const onAddCategory = () => {
        setCategories([ ...categories, 'sailor moon']);
    }

  return (
    <>
       <h1>GifExpertApp</h1>

        <AddCategory />

        <button onClick={ onAddCategory }>Agregar</button>
       <ol>
            {categories.map( category => {
                return <li key={ category }>{category}</li>
            } )}
       </ol>
    </>
  )
}

export default GifExpertApp

