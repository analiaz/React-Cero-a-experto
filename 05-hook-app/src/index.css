body {
    padding: 20px;
}

buttom {
    margin-right: 10px;
}

a {
    margin-right: 5px;
}