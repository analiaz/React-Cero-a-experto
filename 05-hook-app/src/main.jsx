import React from 'react';
import ReactDOM from 'react-dom/client';
//import HooksApp from './HooksApp';
//import {CounterApp} from './01-useState/CounterApp'
//import CounterWhitCustomHook from "./01-useState/CounterWhitCustomHook";

import './index.css';
//import { SimpleForm } from './02-useEffect/SimpleForm';
import {FormWithCustomeHook} from './02-useEffect/FormWithCustomeHook'



ReactDOM.createRoot(document.getElementById('root') ).render(
  <React.StrictMode>
    <FormWithCustomeHook />
  </React.StrictMode>,
)
