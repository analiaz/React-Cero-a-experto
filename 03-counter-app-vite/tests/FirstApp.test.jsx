import { render } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Prueba en <FristApp />', () => {
  
    test('debe de hacer match von el snapshot', () => {

        const title = 'Hola, Soy Goku';
        const {container} = render( <FirstApp title={ title } />);

        expect( container).toMatchSnapshot();

    });

})
