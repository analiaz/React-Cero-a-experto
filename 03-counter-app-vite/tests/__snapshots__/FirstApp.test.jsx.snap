// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <FristApp /> debe de hacer match von el snapshot 1`] = `
<div>
  <h1>
     
    Hola, Soy Goku
     
  </h1>
  <p>
     
    No hay subtitulo
     
  </p>
  <p>
     
    Analia Zunino
     
  </p>
</div>
`;
